lib/modules/api/index.js

```diff
@@ -2,6 +2,105 @@ module.exports = Api;
 
 function Api() {}
 
+Api.prototype.start = async function($happn) {
+  this.domain = $happn._mesh.config.domain || $happn._mesh.config.name;
+  this.__componentAndMethodCache = {};
+  await this.__attachAllExchangeRequests($happn);
+};
+
+Api.prototype.stop = async function($happn) {
+  this.__componentAndMethodCache = {};
+  try {
+    await this.__detachAllExchangeRequests($happn);
+  } catch (e) {
+    $happn.log.warn(`error detaching api component: ${e.message}`);
+  }
+};
+
+Api.prototype.__componentExists = function($happn, component, method) {
+  const possibleEndpointPaths = [
+    `/${$happn._mesh.config.name}/${component}/${method}`,
+    `/${$happn._mesh.config.domain}/${component}/${method}`,
+    `/${component}/${method}`
+  ];
+  if (
+    $happn._mesh.exchange[possibleEndpointPaths[0]] ||
+    $happn._mesh.exchange[possibleEndpointPaths[1]] ||
+    $happn._mesh.exchange[possibleEndpointPaths[2]]
+  ) {
+    //we found the whole endpoint - return true
+    return true;
+  }
+  // endpoint does not exist, we check for a component match
+  // (a missing method will be handled on the component-instance)
+
+  // first create the component paths, so we dont do splits slices and joins
+  // for every endpoint key we are examining
+  const possibleComponentPaths = [
+    `/${$happn._mesh.config.name}/${component}`,
+    `/${$happn._mesh.config.domain}/${component}`,
+    `/${component}`
+  ];
+  return (
```

Comment:
IS there a clearer way of doing this?

Reply:
I'll have to think about it, one thing we may need to look at is caching here though, as this may get expensive...